src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -91,6 +92,21 @@ public class SocketOutput : ISocketOutput
                 if (buffer.Count > 0)
                 {
                     var tail = ProducingStart();
+                    if (tail.IsDefault)
+                    {
+                        if (cancellationToken.CanBeCanceled)
+                        {
+                            return TaskUtilities.GetCancelledTask(
+                                cancellationToken.IsCancellationRequested ?
+                                cancellationToken :
+                                new CancellationToken(true));
```

Comment:
(dupe response) But the write has been cancelled? And if a token is passed in that `CanBeCanceled` it means something is listening for cancellation and will handle it.

Alternative is just to mark as complete, even though it has been cancelled?